cmd/agent/common/helpers.go

```diff
@@ -86,19 +86,19 @@ func SelectedCheckMatcherBuilder(checkNames []string, minInstances uint) func(co
 // SetupInternalProfiling is a common helper to configure runtime settings for internal profiling.
 func SetupInternalProfiling(cfg config.ConfigReader, configPrefix string) {
 	if v := cfg.GetInt(configPrefix + "internal_profiling.block_profile_rate"); v > 0 {
-		if err := settings.SetRuntimeSetting("runtime_block_profile_rate", v, settings.SourceConfig); err != nil {
+		if err := settings.SetRuntimeSetting("runtime_block_profile_rate", v, config.SourceAgentRuntime); err != nil {
```

Comment:
Isn't the value coming from the config, not something changed by a user at runtime using the CLI?